### PR TITLE
feat: Add button types

### PR DESCRIPTION
**Pull request by iRoachie**
Implements #1525

Adds three button types to the Button component:
- solid
- clear
- outline

### Changes
- Deprecates `clear` prop.
- Adds new disabled color to colors

#### Normal state
normal

#### Raised state
raised

#### Disabled state
disabled

### CODE REVIEW

**Comment by iRoachie** [comment]
Still to add tests, and screen shots to docs.

**Comment by Monte9** [comment]
Looks good. And I assume `raised` is a boolean prop.

Can you also share code snippets for those screenshots? would help to see what props you are using to achieve that.

File: package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "logo": "https://opencollective.com/react-native-elements/logo.txt"
   },
   "dependencies": {
+    "color": "^3.1.0",
```
Comment (Monte9) [inline package.json]: what does this packages provide us?

Comment (iRoachie) [reply package.json]: It's a module to work with colors as strings and perform operations like lighten, darken that you'd normally have with sass etc. 

Used it here https://github.com/react-native-training/react-native-elements/pull/1540/files#diff-a0a6368880417eca5ad26c484e9ded7aR215

**Comment by iRoachie** [comment]
Snippets are super simple. Used the theme provider.

```js
const theme = {
  Button: {
    // disabled: true,
    // raised: true,
    buttonStyle: {
      width: 150,
    },
    containerStyle: {
      margin: 16,
    },
  },
};

```

```jsx
<ThemeProvider theme={theme}>
  <Button title="Solid" />
  <Button type="clear" title="Clear" />
  <Button type="outline" title="Outline" />
</ThemeProvider>
```

**Pull request merged**